native/sqlparser_ex/src/lib.rs

```diff
@@ -19,40 +19,85 @@ struct Response {
     message: String,
 }
 
+impl Response {
+    fn ok(message: String) -> Self {
+        Self {
+            status: atoms::ok(),
+            message,
+        }
+    }
+
+    fn error(message: &str) -> Self {
+        Self {
+            status: atoms::error(),
+            message: message.to_string(),
+        }
+    }
+}
+
 #[rustler::nif]
 fn parse(dialect_str: &str, query: &str) -> NifResult<Response> {
     let result = match dialect_str {
         "bigquery" => Parser::parse_sql(&BigQueryDialect {}, query),
+        "clickhouse" => Parser::parse_sql(&ClickHouseDialect {}, query),
         "postgres" => Parser::parse_sql(&PostgreSqlDialect {}, query),
-        _ => Err(ParserError(
-            "Parser for this dialect is not supported.".to_string(),
-        )),
+        _ => return Ok(Response::error("Parser for this dialect is not supported.")),
```

Comment:
a bit apprehensive on this approach, because now any nif errors results in it becoming log messages.
The error message sometimes contains the raw query, which may leak and expose logic to end users as the error message bubbles up via API.
Not against this though as we can properly handle the errors in the elixir layer, but that would mean that we need to sanitize the error message that gets returned to the user instead of showing the SQL string.
Would need to add that into parser.ex or sql.ex to clean the message before we merge this.

Reply:
I'll just go back to the old approach to keep it simple for now. Was mostly using this as a crutch to understand problems during the transition. We can revisit a better approach later.